File: backend/src/main/java/com/nexus/backend/service/InterestService.java
package com.nexus.backend.service;

import com.nexus.backend.dto.request.AddInterestRequest;
import com.nexus.backend.dto.request.UpdateInterestRequest;
import com.nexus.backend.dto.response.InterestResponse;

import java.util.List;

public interface InterestService {

    InterestResponse addInterest(AddInterestRequest request);

    List<InterestResponse> getMyInterests();

    InterestResponse updateInterest(Long id, UpdateInterestRequest request);

    void deleteInterest(Long id);

}

File: frontend/src/services/interestService.js
import api from "./api";

export const getMyInterests = async () => {
  const response = await api.get("/interests");
  return response.data;
};

export const addInterest = async (interestData) => {
  const response = await api.post("/interests", interestData);
  return response.data;
};

export const updateInterest = async (id, interestData) => {
  const response = await api.put(`/interests/${id}`, interestData);
  return response.data;
};

export const deleteInterest = async (id) => {
  await api.delete(`/interests/${id}`);
};